src/java.net.http/share/classes/java/net/http/HttpResponse.java

```diff
@@ -748,6 +749,34 @@ public static <T> BodyHandler<T> buffering(BodyHandler<T> downstreamHandler,
                      .buffering(downstreamHandler.apply(responseInfo),
                                 bufferSize);
          }
+
+        /**
+         * {@return a {@code BodyHandler} limiting the number of bytes consumed
+         * and passed to the given downstream {@code BodyHandler}}
+         * <p>
+         * If the number of bytes received exceeds the maximum number of bytes
```

Comment:
Although the `BodyHandlers` and `BodySubscribers` although note that they are applicable for reponse bodies, should we be adding any clarification in this new API, stating that the limit applies only to the body after the headers are read (however big those may be in bytes)?

Reply:
Maybe we could talk of "body bytes".

Reply:
I think that's a good idea. It isn't too elaborate and yet conveys what this new method is meant for.

Reply:
Fixed in 23f79a20476f784058801454a28d37f37df92982.